### PR TITLE
Use CoordsXY for Paint rotations

### PR DESCRIPTION
**Pull request by tupaschoal**
Helps #10267

### CODE REVIEW

File: src/openrct2/paint/Paint.cpp [hunk 1]
```diff
@@ -891,9 +890,9 @@ paint_struct* sub_98197C(
 
     session->LastRootPS = ps;
 
-    LocationXY16 attach = { (int16_t)ps->bounds.x, (int16_t)ps->bounds.y };
+    CoordsXY attach = { ps->bounds.x, ps->bounds.y };
 
-    rotate_map_coordinates(&attach.x, &attach.y, session->CurrentRotation);
+    attach = attach.Rotate(session->CurrentRotation);
```
Comment (duncanspumpkin) [inline src/openrct2/paint/Paint.cpp]: You could just do `CoordsXY attach = CoordsXY{ ps->bounds.x, ps->bounds.y }.Rotate(session->CurrentRotation);`
Not sure what is better readability wise.

**Comment by duncanspumpkin** [comment]
Was there any measuarble decrease in performance with this change? This is a function that is called a lot.

File: src/openrct2/paint/Paint.cpp (hunk 1 above)
Comment (tupaschoal) [reply src/openrct2/paint/Paint.cpp]: I think this line is small enough for this to be ok, so I'm following your suggestion.

**Comment by tupaschoal** [comment]
> Was there any measuarble decrease in performance with this change? This is a function that is called a lot.

I didn't measure, really. How should I?

**Comment by duncanspumpkin** [comment]
There is a benchmark command but I can't remember it. @ZehMatt could you check to see if there is any change in performance?

**Comment by ZehMatt** [comment]
I can and I did,
develop:
```
>openrct2 benchgfx parks/bpb.sv6
Engine: Software
Render Count: 60
Zoom[0] average: 0.059566s, 17 FPS
Zoom[1] average: 0.027072s, 37 FPS
Zoom[2] average: 0.012890s, 78 FPS
Total average: 0.033176s, 30 FPS
Time: 1.99055s
```
rotate-paint:
```
>openrct2 benchgfx parks/bpb.sv6
Engine: Software
Render Count: 60
Zoom[0] average: 0.059929s, 17 FPS
Zoom[1] average: 0.027087s, 37 FPS
Zoom[2] average: 0.012765s, 78 FPS
Total average: 0.033260s, 30 FPS
Time: 1.99562s
```
No big difference really, super happy that benchgfx is now reliable, not so happy about CoordsXY::Rotate, I think there should be two versions such as: ```CoordsXY CoordsXY::GetRotated``` and ```void CoordsXY::Rotate```, the temporary should be avoided imho but not for this PR obviously.

**Pull request merged**